src/routers/user.ts

```diff
@@ -1,80 +1,118 @@
+    try {
+      const { id } = req.params;
+      const result = await userService.getUserById(id);
+      result
+        ? res.status(200).json({ user: result })
+        : res.status(404).json({ message: `no users with id ${id}` });
+    } catch (e) {
+      res
+        .status(500)
+        .json({ message: e instanceof Error ? e.message : 'Unknown Error' });
+    }
   }
-});
+);
 
 user.put(
   '/user/:id',
   userBodyValidatorOnUpdate,
-  userParamsIdValidator,
-  async (req, res) => {
-    const { id } = req.params;
-    const result = await userService.updateUser(id, req.body);
-    if (result instanceof Error) {
-      res.status(500).json({ message: result });
-    } else {
+  paramsIdValidator,
+  async (
+    req: ValidatedRequest<ParamsIDSchema & UpdateUserBodySchema>,
+    res: Response
+  ) => {
+    try {
+      const { id } = req.params;
+      const result = await userService.updateUser(id, req.body);
       result
         ? res.status(200).json({ updatedUser: result })
         : res.status(404).json({ message: `no users with id ${id}` });
+    } catch (e) {
+      res
+        .status(500)
+        .json({ message: e instanceof Error ? e.message : 'Unknown Error' });
     }
   }
 );
 
-user.delete('/user/:id', userParamsIdValidator, async (req, res) => {
-  const { id } = req.params;
-  const result = await userService.deleteUser(id);
-  if (result instanceof Error) {
-    res.status(500).json({ message: result });
-  } else {
-    result
-      ? res
-          .status(200)
-          .json({ message: `user with id ${id} marked as deleted` })
-      : res.status(404).json({ message: `no users with id ${id}` });
+user.delete(
+  '/user/:id',
+  paramsIdValidator,
+  async (req: ValidatedRequest<ParamsIDSchema>, res: Response) => {
+    try {
+      const { id } = req.params;
+      const result = await userService.deleteUser(id);
+      result
+        ? res
+            .status(200)
+            .json({ message: `user with id ${id} marked as deleted` })
+        : res.status(404).json({ message: `no users with id ${id}` });
+    } catch (e) {
+      res
+        .status(500)
+        .json({ message: e instanceof Error ? e.message : 'Unknown Error' });
+    }
   }
-});
+);
 
-user.get('/users', userQuerySubstringLimitValidator, async (req, res) => {
-  const substring = req.query.login_substring as string;
-  const limit = req.query.limit as string;
+user.get(
+  '/users',
+  userQuerySubstringLimitValidator,
+  async (req: ValidatedRequest<QuerySubstringLimitSchema>, res: Response) => {
+    try {
+      const { login_substring } = req.query;
+      const { limit } = req.query;
 
-  const result = await userService.getUsers(limit, substring);
-  
-  if (result instanceof Error) {
-    res.status(500).json({ message: result });
-  } else {
-    res.status(200).json({ users: result });
+      const result = await userService.getUsers(
+        login_substring as string | undefined,
+        limit as string | undefined
```

Comment:
```suggestion
        login_substring, limit
```
You don't need to explicitly type arguments here after you validated your query. Also why queryparam with underscore?

Reply:
fixed